File: src-tauri/src/commands/runbook.rs
use crate::models::result::CommandResult;
use crate::services::runbook_service::{self, RunbookModelData};

#[tauri::command]
pub async fn get_runbook_model() -> CommandResult<RunbookModelData> {
    match runbook_service::get_runbook_model().await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/runbook.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import { getRunbookModel } from "../services/runbook-service.js";

export function registerRunbookHandlers(): void {
  ipcMain.handle(IPC.GET_RUNBOOK_MODEL, async () => {
    try { return ok(await getRunbookModel()); }
    catch (error) { return err(toAppError(error)); }
  });
}
